lib/service/storage/elasticsearch.js

```diff
@@ -210,6 +210,51 @@ class ElasticSearch extends Service {
       .catch(error => this._esWrapper.reject(error));
   }
 
+  /**
+   * Returns detailed multi-level storage stats data
+   *
+   * @returns {Promise.<Object>}
+   */
+  async stats () {
+    const esRequest = {
+      metric: ['docs', 'store'],
+    };
+
+    const { body } = await this._client.indices.stats(esRequest);
+    const indexes = {};
+    let sizeInBytes = 0;
+
+    for (const [esIndex, esIndexInfo] of Object.entries(body.indices)) {
+      const [ indexName, collectionName ] = esIndex
+        .substr(1, esIndex.length)
+        .split(NAME_SEPARATOR);
+
+      if ( esIndex[0] !== this._indexPrefix || collectionName === HIDDEN_COLLECTION ) {
+        continue;
+      }
+
+      if (!indexes[indexName]) {
+        indexes[indexName] = {
+          collections: [],
+          name: indexName,
+          sizeInBytes: 0,
+        };
+      }
+      indexes[indexName].collections.push({
+        documentCount: esIndexInfo.total.docs.count,
+        name: collectionName,
+        sizeInBytes: esIndexInfo.total.store.size_in_bytes,
+      });
+      indexes[indexName].sizeInBytes += esIndexInfo.total.store.size_in_bytes;
+      sizeInBytes += esIndexInfo.total.store.size_in_bytes;
+    }
+
+    return {
+      indexes: Array.from(Object.values(indexes)),
```

Comment:
Object.values return an array already
```suggestion
      indexes: Object.values(indexes),
```